zipline/assets/assets.py

```diff
@@ -343,14 +352,16 @@ def _retrieve_assets(self, sids, asset_tbl, asset_type):
             return {}
 
         cache = self._asset_cache
-
         hits = {}
-        # Load misses from the db.
-        query = self._select_assets_by_sid(asset_tbl, sids)
-        for row in imap(dict, query.execute().fetchall()):
-            asset = asset_type(**_convert_asset_timestamp_fields(row))
-            sid = asset.sid
-            hits[sid] = cache[sid] = asset
+
+        for chunk in self._group_into_chunks(list(sids)):
```

Comment:
Since both callers `list` the arg first, you could move that into the group function.